test/linalg/dense.jl

```diff
@@ -582,6 +582,25 @@ end
     end
 end
 
+@testset "issue #23366 (Int Matrix to Int power)" begin
+    @testset "Tests for $elty" for elty in (Int128, Int16, Int32, Int64, Int8,
+                                            UInt128, UInt16, UInt32, UInt64, UInt8,
+                                            BigInt)
+        @test_throws DomainError elty[1 1;1 0]^-2
+        @test elty[1 1;1 0]^2 == elty[2 1;1 1]
+        I_ = elty[1 0;0 1]
+        @test I_^-1 == I_
+        if !(elty<:Unsigned)
+            @test (-I_)^-1 == -I_
+            @test (-I_)^-2 == I_
+        end
+        for elty2 = (Int64, BigInt)
+            TT = Base.promote_op(^, elty, elty2)
+            @test Base.return_types(^, (Array{elty,2}, elty2)) == [Array{TT,2}]
```

Comment:
Okay nice, perhaps worth adding a comment about that so it does not get removed by someone later on. Perhaps this could otherwise be tested with something like
```julia
@test ([1 1; 1 0]^big(1))::Matrix{BigInt} == [1 1; 1 0]
```
such that we test the return value and not just the type? :)